appinventor/components/src/com/google/appinventor/components/runtime/util/TextViewUtil.java

```diff
@@ -168,13 +168,11 @@ public static Typeface getTypeFace(Form form, String fontFile) {
     Typeface typeface;
     if (!fontFile.contains("/")) {
       if (form instanceof ReplForm) {
-        if (Build.VERSION.SDK_INT > 28) {
-          File file = new File("/storage/emulated/0/Android/data/edu.mit.appinventor.aicompanion3/files/assets/" + fontFile);
-          typeface = Typeface.createFromFile(file);
-        } else {
-          File file = new File("/storage/emulated/0/AppInventor/assets/" + fontFile);
-          typeface = Typeface.createFromFile(file);
-        }
+        try {
+          File file = new File(MediaUtil.fileUrlToFilePath(form.getAssetPath(fontFile)));
+        } catch (IllegalArgumentException e) {
+        } catch (Exception e) {}
+        typeface = Typeface.createFromFile(file);        
```

Comment:
This does not seem like it is legal here as file is defined within the scope of the try. Did you compile and test this?

Reply:
I didn't compile the source, I tested this piece of code by simulating it in an extension.  When I called fileUrlToFilePath() the compiler returned an error without try{}catch{}.

Reply:
That probably wasn't the best way to test. At the moment, this PR will not compile due to this line.